azkaban-common/src/main/java/azkaban/project/NodeBeanLoader.java

```diff
@@ -86,15 +90,84 @@ public AzkabanNode toAzkabanNode(final NodeBean nodeBean) {
     }
   }
 
+  private void validateSchedule(final FlowTriggerBean flowTriggerBean) {
+    final Map<String, String> scheduleMap = flowTriggerBean.getSchedule();
+
+    Preconditions.checkNotNull(scheduleMap, "flow trigger schedule must not be null");
+
+    Preconditions.checkArgument(scheduleMap.containsKey("type") && scheduleMap.get("type")
+        .equals("cron"), "flow trigger schedule type must be cron");
+
+    Preconditions.checkArgument(scheduleMap.containsKey("value") && CronExpression
+        .isValidExpression(scheduleMap.get("value")), "flow trigger schedule value must be a "
+        + "valid cron expression");
+
+    Preconditions.checkArgument(scheduleMap.size() == 2, "flow trigger schedule must "
+        + "contain type and value only");
+  }
+
+  private void validateFlowTriggerBean(final FlowTriggerBean flowTriggerBean) {
+    // validate max wait mins
+    Preconditions.checkArgument(flowTriggerBean.getMaxWaitMins() >= 1, "max wait min must be "
```

Comment:
Define the numbers "2" and "1" in constants?

Reply:
put 1 in constants, 2 is just the number of keys to be held, which is not a meaningful number, so i prefer to make it just a hard coded value here.